client/my-sites/earn/memberships/add-edit-plan-modal.jsx

```diff
@@ -205,29 +208,42 @@ const RecurringPaymentsPlanAddEditModal = ( {
 					welcome_email_content: editedCustomConfirmationMessage,
 					subscribe_as_site_subscriber: editedPostsEmail,
 				},
-				translate( 'Updated "%s" product.', { args: editedProductName } )
+				isEnabled( 'earn/rename-payment-blocks' )
+					? translate( 'Updated "%s" payment plan.', { args: editedProductName } )
+					: translate( 'Updated "%s" product.', { args: editedProductName } )
 			);
 		}
 		closeDialog();
 	};
 
 	const renderGeneralTab = () => {
+		const isPayments = isEnabled( 'earn/rename-payment-blocks' );
+		const editProduct = isPayments
+			? translate( 'Edit your existing payment plan.' )
+			: translate( 'Edit your existing Recurring Payments plan.' );
+		const noProduct = isPayments
+			? translate(
+					'Each amount you add will create a separate payment plan. You can create many of them.'
+			  )
+			: translate(
+					'Each amount you add will create a separate Recurring Payments plan. You can create multiple plans.'
+			  );
 		return (
 			<>
-				<p>
-					{ product
-						? translate( 'Edit your existing Recurring Payments plan.' )
-						: translate(
-								'Each amount you add will create a separate Recurring Payments plan. You can create multiple plans.'
-						  ) }
-				</p>
+				<p>{ product ? editProduct : noProduct }</p>
 				<FormFieldset>
 					<FormLabel htmlFor="currency">{ translate( 'Select price' ) }</FormLabel>
 					{ product && (
 						<Notice
-							text={ translate(
-								'Updating the price will not affect existing subscribers, who will pay what they were originally charged.'
-							) }
+							text={
+								isPayments
+									? translate(
+											'Updating the price will not affect existing subscribers, who will pay what they were originally charged on renewal.'
```

Comment:
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.